### PR TITLE
FlameGraph: Optimize rendering of muted regions

### PR DESCRIPTION
**Pull request by leeoniya**
this improves flamegraph rendering performance by 45% in graphs with thousands of tiny `muted` microregions of the same color. instead of filling each individually we fill them all at once after building up the path in a Path2D object. this all but eliminates the top item in the profile.

- still need to pick the correct static color for light and dark mode, i just hardcoded `#666`.
- looks like there's still a bug with the previous Path2D not getting invalidated when switching from Both to FlameGraph-only view. this clears up when you focus a block. probably some React hook thing :shrug: 
- only tested component in a dashboard (not Explore or Pyroscope app)

before:

image

after:

image

### CODE REVIEW

File: packages/grafana-flamegraph/src/FlameGraph/rendering.ts [hunk 1]
```diff
@@ -277,14 +337,24 @@ export function walkTree(
...
-      renderFunc(item, barX, barY, width, height, label, muted);
+      renderObjects.push({
+        item,
+        x: barX,
+        y: barY,
+        width,
+        height,
+        label,
+        muted,
+      });
```
Comment (leeoniya) [inline packages/grafana-flamegraph/src/FlameGraph/rendering.ts:348]: this makes me nervous. it will allocate possibly 200k objects on every resize/redraw?

**Comment by leeoniya** [comment]
LGTM :shipit:

**Review by petethepig — APPROVED** [review]
LGTM

File: packages/grafana-flamegraph/src/FlameGraph/rendering.ts [hunk 2]
```diff
@@ -108,62 +155,66 @@ type RenderFunc = (
...
         const numberOfCollapsedItems = collapsedItemConfig.items.length;
-        label = `(${numberOfCollapsedItems}) ` + label;
+        finalLabel = `(${numberOfCollapsedItems}) ` + label;
       }
 
-      if (muted) {
-        // Only fill the muted rects
-        ctx.fill();
+      if (collapsedItemConfig) {
+        if (width >= LABEL_THRESHOLD) {
```
Comment (joey-grafana) [inline packages/grafana-flamegraph/src/FlameGraph/rendering.ts:196]: nit: `if (width >= LABEL_THRESHOLD)` used in both if and else, should move up

**Review by joey-grafana — APPROVED** [review]

**Pull request merged**